src/engine/objects/robot/mechanisms/SimGripperMechanism.ts

```diff
@@ -0,0 +1,267 @@
+import {
+  Box,
+  PrismaticJoint,
+  Vec2,
+  World,
+  WeldJoint,
+  WeldJointDef,
+  PrismaticJointDef,
+} from "planck-js";
+import { getSensorMountPosition } from "../../../utils/RobotUtils";
+import {
+  IMechanismIOConfig,
+  IMechanismSpec,
+  IRobotSpec,
+  SensorMountingFace,
+  SensorSpec,
+} from "../../../specs/RobotSpecs";
+import { SimMechanism } from "./SimMechanism";
+import * as THREE from "three";
+import { SimGripperJaw } from "./SimGripperJaw";
+import { SimContactSensor } from "../sensors/SimContactSensor";
+import { EntityCategory, EntityMask } from "../RobotCollisionConstants";
+
+enum GripperState {
+  OPEN,
+  CLOSING,
+  OPENING,
+  CLOSED_EMPTY,
+  CLOSED_FULL,
+}
+
+const IO_TYPE_HELD = "held"; // held == 1 -> grabbed something, held == 0 nothing grabbed
+const IO_TYPE_GRAB = "grab"; // set to 1 to start grabbing, set to 0 to open grabber
+
+/**
+ * This mechanism enables the robot to grab objects in the scene.
+ *
+ * The physical model for this mechanism looks like this (top down):
+ *
+ *     |      |
```

Comment:
I really like what you've built here 👍🏻 

Random side note: I had a look into some of the Box2D examples and found an example of "forklift"  http://www.iforce2d.net/b2dtut/joints-prismatic - which can pick things up.  But I think what you have is equally nice (and closer to what we want requirement-wise)

Reply:
yes this is done in a very similar way, the jaw which moves is on a prismatic joint.